src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs

```diff
@@ -118,6 +119,15 @@ private static void OnExit(object sender, EventArgs e)
         Log("OpenTelemetry tracer exit.");
     }
 
+    private static void OnUnhandledException(object sender, UnhandledExceptionEventArgs args)
```

Comment:
In the future we could consider attempting to report this exception. I do not believe that we can appropriately record the exception as an event on a span, because the necessary span context might not be available.

Reply:
I think it is possible something more or less like:

```csharp
if (Activity.Current != null) { 
  // we can report an even or make set error span status as and add the description from exception message
}
```

WDYT?

Reply:
I have just found https://github.com/open-telemetry/opentelemetry-dotnet/blob/d716bb17cd8115a3c6d7afe6da74652000aa8372/docs/trace/reporting-exceptions/README.md#unhandled-exception

Reply:
Even if we use code like that we would still potentially lose some of the spans that are inflight from parallel work, but it would potentially provide some additional context for resolving the error if the span context is available. At the very least, the code that we have needs to be guarded better. We really should prevent exceptions from being thrown in event handlers.

Reply:
This can be addressed in a separate PR if you want

Reply:
Created https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/390